synapse/handlers/stats.py

```diff
@@ -135,13 +131,12 @@ def _unsafe_process(self):
 
             self.pos = max_pos
 
-    @defer.inlineCallbacks
-    def _handle_deltas(self, deltas):
+    async def _handle_deltas(self, deltas):
         """Called with the state deltas to process
 
         Returns:
-            Deferred[tuple[dict[str, Counter], dict[str, counter]]]
-            Resovles to two dicts, the room deltas and the user deltas,
+            tuple[dict[str, Counter], dict[str, counter]]
+            Resolves to two dicts, the room deltas and the user deltas,
```

Comment:
ftr, "Resolves" only really makes sense when we're talking about deferreds
```suggestion
            Two dicts: the room deltas and the user deltas,
```